clang/include/clang/AST/Expr.h

```diff
@@ -3518,6 +3523,13 @@ class CompoundLiteralExpr : public Expr {
     TInfoAndScope.setPointer(tinfo);
   }
 
+  bool hasStaticStorage() const { return isFileScope() && isGLValue(); }
+  APValue *getOrCreateStaticValue(ASTContext &Ctx) const;
+  APValue &getStaticValue() const {
+    assert(StaticValue);
```

Comment:
Suggestion: an assertion macro in a header is asking for ODR violations. Could we move it to `.cpp` file?
PS I didn't check how clean LLVM is from those, maybe I'm too aggressive with this.